### DISCUSSION TITLE
Channels with localhost als repository server

### DISCUSSION THREAD

**Discussion by crefeldtc** [uyuni-project/uyuni#8711] [Q&A]
Hi,

since today's upgrade from 2023.12 to 2024.03 Uyuni is changing the hostname of all baseurls of "susemanager:channels,repo" from lkv-srvapp39.lkvbay.net:443 (which is correct) to localhost:443. This happens to a system whenever I mark a system and run "Apply patches". Of course this means that the patch procedure is failing

This means that only systems that haven't been selected for patching by Uyuni can still run locally "zypper up".

Any idea how to correct this?

Regards, 
 Tobias Crefeld.

**Comment by mcalmer** [comment]
This sounds like something wrong when we move uyuni to containers.
We have to change the hostname for all internal operations to localhost, and introduced a new hostname config option to reflect the external hostname. 
@cbosdo @mbussolotto is there a known bug?

**Reply by cbosdo** [reply]
What looks strange is that this value is taken from the pillar data generated in [MinionGeneralPillarGenerator](https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/suse/manager/webui/services/pillar/MinionGeneralPillarGenerator.java#L134) and the host is coming from the [Server](https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/domain/server/Server.java#L2347).

Unless the `java.hostname` value has been changed to `localhost` in `/etc/rhn/rhn.conf` (which I doubt) I don't understand where the value comes from.

**Reply by crefeldtc** [reply]
The file `/etc/rhn/rhn.conf` has a time stamp from 2022, so probably it hasn't changed by the update of Uyuni some days ago.
I find no entry `java.hostname` in this file. Should I add it?
The only entry with a value "localhost" is `db_host`.

**Reply by mbussolotto** [reply]
> I find no entry java.hostname in this file. 

I think this might explains the issue. Probably it was not visible before because for some configuration we (wrongly) used `cobbler.host` instead of `java.hostname`. This problem was fixed by https://github.com/uyuni-project/uyuni/commit/6d732e14a2c8256a60277757f274d8aa05516db6 and it was introduced in 2024.03 . I'm not sure if we should investigating why `java.hostname` is missing, but for sure setting `java.hostname` to FQDN might fix your issue.

**Reply by crefeldtc** [reply]
I added this entry and rebooted the system that runs Uyuni without success:
```
lkv-srvapp39:~ # grep java /etc/rhn/rhn.conf
# taskomatic.java.maxmemory=4096
java.salt_check_download_tokens = false
java.hostname = lkv-srvapp39.lkvbay.net
```
It is still replacing the hostname of the baseurl by `localhost` .

Are the baseurls created on the fly when I press "Apply patches"? 
Is there another process, maybe during the update of Uyuni that is creating the baseurls and store it somewhere?

**Reply by mcalmer** [reply]
After changing the config, you need to restart the services and maybe you need to regenerate the pillar data. Try to go to a system and select the properties page. Click on the Update button. That should do it.

**Reply by cbosdo** [reply]
@mbussolotto @mcalmer didn't we add an update script to add that property when updating the package?

**Reply by mbussolotto** [reply]
> didn't we add an update script to add that property when updating the package?

I don't think an update script is required here and would fix the problem. Generally the update script should change a param if it was set with the default value and there's new default value. In this case the default values are not changed (`java.hostname` should be the FQDN and since it's not a uyuni container `cobbler.host` can remain FQDN). Moreover, it wouldn't work because `java.hostname` was missing.

OTOH we have an update script in `mgradm` for `cobbler.host`, since the default value for container should be `localhost` instead of `FQDN` (for non container, `cobbler.host` can be `FQDN` or `localhost`, there's no difference between the 2 values).

The real problem for me here was `java.hostname` missing,I'm not sure how it can happened

**Reply by mcalmer** [reply]
In the past it did not exist. In 4.3 we do not have this. So the question is, during a migration how to add it?
In Uyuni, we just update the packages. I think we need a %post script which set the hostname when cobbler.host is set to localhost.
I think the problem is, that the pillar expect now java.hostname exist, while this only exist in the container?

**Reply by crefeldtc** [reply]
Rebooting Uyuni restarted the services, but this didn't help or wasn't enough.
At last updating the properties was successful. I'm glad that we have only twenty systems as there is no SSM feature or something similar to update the properties of all system in one shot.
For now it seems that everything's is running fine.
Thanks for your assistance!

**Reply by mbussolotto** [reply]
ohh thanks for the explanation, I didn't notice `java.hostname` was introduced recently https://github.com/uyuni-project/uyuni/pull/6216 . 

> I think the problem is, that the pillar expect now java.hostname exist, while this only exist in the container?

I think `java.hostname` doesn't exist just for fresh installation before https://github.com/uyuni-project/uyuni/pull/6216. 

> I think we need a %post script which set the hostname when cobbler.host is set to localhost.

Maybe we can just call `spacewalk-hostname-rename` on %post script ?

**Reply by crefeldtc** [reply]
I found a file `/etc/rhn/rhn.conf.orig` that was created during the initial installation of Uyuni in March 2020.
It contains `cobbler.host = lkv-srvapp39.lkvbay.net` but no `java.hostname` .

I think that the baseurl of all clients changed during the upgrade from 2023.12 to 2024.03 to `localhost `, not only the systems that I tried to patch after the upgrade and before adding `java.hostname`. The only difference was that the file  `/etc/zypp/repos.d/susemanager\:channels.repo` of the other systems still contained the FQDN. But even after adding `java.hostname` and restarting the services a patch by Uyuni of the respective clients replaced the FQDN by `localhost` .

**Comment by crefeldtc** [comment]
Hi Michael,
thanks for your feedback!
Do you think that there is a workaround to deal with this issue?
Otherwise it's probably better to return to the system's snapshot before the upgrade.
Regards,
 Tobias Crefeld.

**Reply by mcalmer** [reply]
I hope  in a few days we will release the next uyuni version. I would give it a try. I think it should be fixed.

**Comment by crefeldtc** [comment]
Thanks for you reply!
This will be a regular minor update, right? So I guess, it should make no difference, if I go back to 2023.03 meanwhile and try the next upgrade as soon as it's available.
Regards,
 Tobias Crefeld.
